app/src/main/java/com/example/leagueoflegends/view/SplashActivity.kt

```diff
@@ -0,0 +1,42 @@
+package com.example.leagueoflegends.view
+
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.os.Handler
+import com.example.leagueoflegends.R
+
+class SplashActivity : AppCompatActivity() {
+    private var mDelayHandler: Handler? = null
+    private val SPLASH_DELAY: Long = 3000 //3 seconds
+
+    internal val mRunnable: Runnable = Runnable {
+        if (!isFinishing) {
+
+            val intent = Intent(applicationContext, MainActivity::class.java)
+            startActivity(intent)
+            finish()
+        }
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_splash)
+
+        //Initialize the Handler
+        mDelayHandler = Handler()
+
+        //Navigate with delay
+        mDelayHandler!!.postDelayed(mRunnable, SPLASH_DELAY)
```

Comment:
avoid using exclamation marks and solve it with ?.let{} instead. Null Pointer exception danger